frontend/packages/kubevirt-plugin/src/components/virtualization-overview/RunningVmsPerTemplateCard.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { ChartDonut, ChartLabel, ChartLegend } from '@patternfly/react-charts';
+import { Card, CardBody, CardHeader, CardTitle } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { VirtualMachineModel } from '../../models';
+import { kubevirtReferenceForModel } from '../../models/kubevirtReferenceForModel';
+import { VMKind } from '../../types';
+
+const getVmsPerTemplateMap = (vms, vmsLoaded) => {
+  let numVMs = 0;
+  const map = {};
+  vmsLoaded &&
+    vms.forEach((vm) => {
+      const template = vm?.metadata?.labels['vm.kubevirt.io/template'];
+      map[template] = map.hasOwnProperty(template) ? map[template] + 1 : 1;
+      numVMs++;
+    });
+
+  const vmCount = [];
+  const legendLabel = [];
+  const dataMap = [];
+
+  for (const templateName in map) {
+    if (Object.prototype.hasOwnProperty.call(map, templateName)) {
+      const count = map[templateName];
+      const percent = Math.round((count / numVMs) * 100);
+      dataMap.push({ x: templateName, y: percent });
+      vmCount.push(count);
+      legendLabel.push({ name: `${count}: ${templateName}` });
+    }
+  }
+
+  return { data: vmCount, legend: legendLabel, map: dataMap };
+};
+
+const LegendLabel = ({ values, ...rest }) => (
+  <ChartLabel {...rest} style={[{ fontSize: '10' }]} text={rest.text} />
```

Comment:
optional nitpick: I'm not a fan of inline CSS styles to be honest, any chance to export this to a classname?

Reply:
Done!